### PR TITLE
IBP Loss Weighting

### LINKED ISSUE #2102: Loss Weighting Overridden in IBP Training
**Describe the bug**
If the user chooses not to use a scheduler for the loss weighting, and wants to use a fixed weighting, their supplied value is overridden and a default one is used instead.

**To Reproduce**
1. Go to ibp_certified_trainer_pytorch.py
2. Scroll down to line 304
4. loss_weighting_k = 0.1 should be loss_weighting_k = self.loss_weighting

**System information (please complete the following information):**
 - OS: Mac OS
 - Python version: Python 3.8
 - ART version or commit number: 1.14.0
 - PyTorch 1.13.1

### PR DESCRIPTION
**Pull request by GiulioZizzo**
# Description

We fix an issue where the default value for a loss weighting parameter was being used rather than user supplied inputs in certain situations. Further, we add additional input checking to the ibp trainer. 

Fixes  #2102

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Existing tests pass

**Test Configuration**:
- OS: Mac OS
- Python version: 3.8
- ART version or commit number: 1.14
- Pytorch 1.13.1

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by beat-buesser — APPROVED** [review]
Hi @GiulioZizzo Thank you very much, looks good to me.
